src/com/goide/inspections/GoUnusedImportDeclaration.java

```diff
@@ -62,6 +69,19 @@ protected void checkFile(@NotNull GoFile file, @NotNull ProblemsHolder problemsH
     for (GoImportSpec duplicatedImportSpec : GoImportOptimizer.findDuplicatedEntries(importMap)) {
       problemsHolder.registerProblem(duplicatedImportSpec, "Redeclared import", ProblemHighlightType.GENERIC_ERROR, OPTIMIZE_QUICK_FIX);
     }
+
+    // Find imports that are referred to by the same alias or last pathname component.
+    MultiMap<String, GoImportSpec> nameMap = getImportNameMap(file);
```

Comment:
Should be fixed, now I only suggest a fix if the import wasn't already flagged in the above loop.